controllers/nodes/nodes_controller_test.go

```diff
@@ -131,5 +145,431 @@ var _ = Describe("NodesController", func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updatedNode.Labels).To(HaveKey(labelNodeRoleControlPlane))
+				Expect(updatedNode.Labels[labelNodeRoleControlPlane]).To(Equal(hypershiftLabelValue))
+			})
+
+			It("Should not label control plane node", func() {
+				cpNode := &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "control-plane-1",
+						Labels: map[string]string{
+							labelNodeRoleControlPlane: "",
+						},
+					},
+				}
+
+				hco := commontestutils.NewHco()
+				resources := []client.Object{hco, cpNode}
+				cl := commontestutils.InitClient(resources)
+
+				r := &ReconcileNodeCounter{
+					Client:                       cl,
+					nodeEvents:                   nodeEvents,
+					shouldLabelNodes:             true,
+					HandleHyperShiftNodeLabeling: HandleHyperShiftNodeLabeling,
+				}
+
+				nodeinfo.HandleNodeChanges = func(_ context.Context, _ client.Client, _ *hcov1beta1.HyperConverged, _ logr.Logger) (bool, error) {
+					return false, nil
+				}
+
+				nodeRequest := reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name: "control-plane-1",
+					},
+				}
+
+				res, err := r.Reconcile(context.TODO(), nodeRequest)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(res.IsZero()).To(BeTrue())
+
+				// Verify the node label wasn't changed
+				updatedNode := &corev1.Node{}
+				err = cl.Get(context.TODO(), client.ObjectKey{Name: "control-plane-1"}, updatedNode)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updatedNode.Labels[labelNodeRoleControlPlane]).To(Equal(""))
+			})
+
+			It("Should skip labeling when shouldLabelNodes is false", func() {
+				workerNode := &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "worker-1",
+						Labels: map[string]string{
+							labelNodeRoleWorker: "",
+						},
+					},
+				}
+
+				hco := commontestutils.NewHco()
+				resources := []client.Object{hco, workerNode}
+				cl := commontestutils.InitClient(resources)
+
+				r := &ReconcileNodeCounter{
+					Client:     cl,
+					nodeEvents: nodeEvents,
+					HandleHyperShiftNodeLabeling: func(_ context.Context, _ client.Client, _ string, _ logr.Logger) error {
+						return nil
+					},
+				}
+
+				nodeinfo.HandleNodeChanges = func(_ context.Context, _ client.Client, _ *hcov1beta1.HyperConverged, _ logr.Logger) (bool, error) {
+					return false, nil
+				}
+
+				nodeRequest := reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name: "worker-1",
+					},
+				}
+
+				res, err := r.Reconcile(context.TODO(), nodeRequest)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(res.IsZero()).To(BeTrue())
+
+				// Verify the node was not labeled
+				updatedNode := &corev1.Node{}
+				err = cl.Get(context.TODO(), client.ObjectKey{Name: "worker-1"}, updatedNode)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updatedNode.Labels).NotTo(HaveKey(labelNodeRoleControlPlane))
+			})
+
+			It("Should remove HyperShift label when shouldLabelNodes is false", func() {
```

Comment:
I don't understand this test. It does not check the actual behavior, but adds a fake one, and check it.

Please drop this test.